src/websiteAccount/WebsiteAccount.java

```diff
@@ -297,6 +297,31 @@ public static boolean updateWebsiteAccount(ConnectionSource databaseConnection,
 		}
 		return false;
 	}
-	
-	
+	public static boolean updateWebsiteAccount(ConnectionSource databaseConnection, String currentAccountUsername, User safeStoreUser, String[] fieldsToModify,String[] newInputs) {
+		try {
+			WebsiteAccount accountToModify = WebsiteAccount.getWebsiteAccountFromNickname(databaseConnection,currentAccountUsername, safeStoreUser);
+
+			for(int i = 0; i < fieldsToModify.length; i++) {
+
+				if (fieldsToModify[i].equals("Nickname")) {
+					String newNickname = newInputs[0].trim();
+					accountToModify.setNickname(newNickname);
+				}
+				if (fieldsToModify[i].equals("Login")) {
+					String newLogin = newInputs[1].trim();
+					accountToModify.setWebsiteLogin(newLogin);
+				}
+				if (fieldsToModify[i].equals("Password")) {
+					String newPassword = newInputs[2].trim();
+					accountToModify.setWebsitePassword(newPassword);
+				}
+			}
```

Comment:
I don't think this needs a for loop. How about: 
```suggestion
			if (fieldsToModify[0].equals("Nickname")) {
					String newNickname = newInputs[0].trim();
					accountToModify.setNickname(newNickname);
				}
				if (fieldsToModify[1].equals("Login")) {
					String newLogin = newInputs[1].trim();
					accountToModify.setWebsiteLogin(newLogin);
				}
				if (fieldsToModify[2].equals("Password")) {
					String newPassword = newInputs[2].trim();
					accountToModify.setWebsitePassword(newPassword);
				}
```